src/Altinn.App.Core/Features/Signing/SigningDelegationService.cs

```diff
@@ -26,25 +27,48 @@ CancellationToken ct
             {
                 if (state.IsAccessDelegated is false)
                 {
-                    // csharpier-ignore-start
-                    string appResourceId = AppIdHelper.ToResourceId(instance.AppId);
-                    DelegationRequest delegation = DelegationRequestBuilder
-                        .CreateBuilder(appResourceId, instance.Id)
-                        .WithDelegator(new Delegator { IdType = DelegationConst.Party, Id = "" }) // TODO: assign delegator
-                        .WithRecipient(new Delegatee { IdType = DelegationConst.Party, Id = signeeContext.PartyId.ToString() })
-                        .AddRight()
-                            .WithAction(DelegationConst.ActionId, ActionType.Read)
-                            .AddResource(DelegationConst.Resource, appResourceId) // TODO: translate app id to altinn resource id
-                            .AddResource(DelegationConst.Task, taskId)
-                            .BuildRight()
-                        .AddRight()
-                            .WithAction(DelegationConst.ActionId, ActionType.Sign)
-                            .AddResource(DelegationConst.Resource, appResourceId) // TODO: translate app id to altinn resource id
-                            .AddResource(DelegationConst.Task, taskId)
-                            .BuildRight()
+                    DelegationRequest delegationRequest = DelegationBuilder
+                        .Create()
+                        .WithApplicationId(instance.AppId)
+                        .WithInstanceId(instance.Id)
+                        .WithDelegator(new Delegator { IdType = DelegationConst.Party, Id = "" })
+                        .WithRecipient(
+                            new Delegatee { IdType = DelegationConst.Party, Id = signeeContext.PartyId.ToString() }
+                        )
+                        .WithRights(
+                            [
+                                AccessRightBuilder
+                                    .Create()
+                                    .WithAction(DelegationConst.ActionId, ActionType.Read)
```

Comment:
Do we need the first parameter on `WithAction` where you pass in `DelegationConst.ActionId`? Or can that we hardcoded within the method? Will it not always be the same for any action?

Same for Resource probably. Isn't the type for a Resource always` DelegationConst.Resource`?

Reply:
Resource may have Type = DelegationConst.Task, but it makes sense to set Resource as default with the possibility to override. Action can be simplified as suggested.